### DISCUSSION TITLE
Rework error message when wrong database version?

### DISCUSSION THREAD

**Discussion by curquiza** [meilisearch/product#572] [Feedback & Feature Proposal]
As part of the CLI stabilization but also the error handler rework for v1, I'm wondering how we could improve the error message when launching Meilisearch but the DB version is the wrong one (if we need to!)

Currently, if my data.ms has been created with Meilisearch `v0.29.2` and I try to run Meilisearch v0.30.0 using the same data.ms, I will get the following error

```
Error: Expected Meilisearch engine version: 0.29.2, current engine version: 0.30.0.
To update Meilisearch please follow our guide on https://docs.meilisearch.com/learn/advanced/updating.html.
```

@meilisearch/docs-team I would love your input on this. Is it clear enough from your point of view? How could we improve?
Same, @oluademola, on the support side, did you encounter issues regarding this error message? Or is it clear enough?

**Comment by guimachiavelli** [comment]
I don't think the current message is that bad, but it might benefit from and update so it's better aligned with [the suggestion for the missing master key error](https://github.com/meilisearch/product/discussions/565#discussioncomment-4266375). Also, I think it might be clearer to mention that it's a mismatch between database and engine versions.

`Error: Your database version (v0.29.2) is incompatible with your current engine (v0.30.0). To migrate data between Meilisearch versions, follow our guide on https://docs.meilisearch.com/learn/advanced/updating.html`

**Reply by dureuill** [reply]
Implemented in https://github.com/meilisearch/meilisearch/pull/3279

**Comment by oluademola** [comment]
I agree with Gui. Maybe it's safe to assume the current error message appears clear and understandable to most users, as I haven't received queries or feedback from the support side about it. However, it could be improved, and Gui's suggestion looks good to me.

**Comment by gmourier** [comment]
Hey everyone 👋

I'm locking that discussion since the proposed changes will be released with `v1`.

Thank you all!
